### PR TITLE
feature: tarpit() filter

### PR DESCRIPTION
**Pull request by szuecs**
feature: tarpit() filter to let connection to the client open until it closes the connection

### CODE REVIEW

File: filters/diag/tarpit.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	if !ok {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	d, err := time.ParseDuration(s)
+	if err != nil {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	t.d = d
```
[the remark this answers is not in the document]
Comment (szuecs) [reply filters/diag/tarpit.go:38]: Haha yeah good point 👍

File: filters/diag/tarpit.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/zalando/skipper/filters"
+)
+
+type tarpitSpec struct {
+	d time.Duration
```
Comment (AlexanderYastrebov) [inline filters/diag/tarpit.go:11]: I think this field is unused

Comment (szuecs) [reply filters/diag/tarpit.go:11]: ah yes it's used as setting the default but never possible to use the default.

Comment (szuecs) [reply filters/diag/tarpit.go]: fixed

**Comment by AlexanderYastrebov** [comment]
I think this is equivalent in behavior with
```
$ bin/skipper -inline-routes='* -> chunks(1, "1s") -> dropResponseHeader("Content-Length") -> repeatContent("X", 1000000000) -> status(200) -> <shunt>'
...
~$ nc localhost 9090
GET / HTTP/1.1        
Host: localhost:9090

HTTP/1.1 200 OK
Server: Skipper
Date: Fri, 01 Dec 2023 09:32:40 GMT
Transfer-Encoding: chunked

1
X
1
X
1
X
1
X
1
X
1
X
1
X
1
X
^C
```
-- it responds with `Transfer-Encoding: chunked` and sends chunks of 1 byte.

If we do not drop Content-Length then it just slowly sends response:
```
$ bin/skipper -inline-routes='* -> chunks(1, "1s") -> repeatContent("X", 10000) -> status(200) -> <shunt>'
...
~$ nc localhost 9090
GET / HTTP/1.1 
Host: localhost:9090

HTTP/1.1 200 OK
Content-Length: 10000
Server: Skipper
Date: Fri, 01 Dec 2023 09:33:42 GMT

XXXXXX^C
```

**Comment by szuecs** [comment]
:+1:

**Comment by AlexanderYastrebov** [comment]
:+1:

**Pull request merged**